### PR TITLE
Define CRDs for ServiceRole and ServiceRoleBinding in pilot

### PR DESCRIPTION
**Pull request by yangminzhu**
1. create corresponding CRD for ServiceRole and ServiceRoleBinding (See
https://github.com/istio/api/blob/master/rbac/v1alpha1/rbac.proto).
The ServiceRole and ServiceRoleBinding is currently defined in istio-auth.yaml
but we have plans to migrate them to pilot for the local RBAC.

2. add corresponding validation code for new model types (#3927).

3. in longer term after the local RBAC is implemented, the pilot watches for changes on Istio RBAC policies, fetches the updated RBAC policies from Istio Config Store and distributes relevant Istio RBAC policies to Envoy proxies that are co-located with service instances and enforce the policies inside Envoy (The mixer won't handle the RBAC policies anymore).

See the design doc (WIP) for more details: https://docs.google.com/document/d/1XkN9ls-ZDISz2uXGN20RaIre63DkJJOJNUOzcKCxbWA/edit?usp=sharing

### CODE REVIEW

**Comment by ayj** [comment]
The code LGTM. General question: what does Pilot do with this new configuration, i.e. who is the consumer?

**Comment by yangminzhu** [comment]
Thanks for the review! Currently pilot doesn't handle the new configuration other than the validation (both in admission controller and istioctl). There is a RBAC mixer adapter to consume these actual resources and apply the corresponding RBAC rules. 
We have plans to implement the local RBAC and we'll need to change the pilot to directly consume these configuration at that time.

File: pilot/pkg/model/config.go [hunk 1]
```diff
@@ -477,6 +477,26 @@ var (
 		Validate:    ValidateEndUserAuthenticationPolicySpecBinding,
 	}
 
+	// ServiceRole describes an RBAC service role.
+	ServiceRole = ProtoSchema{
+		Type:        "service-role",
+		Plural:      "service-roles",
+		Group:       "config",
+		Version:     istioAPIVersion,
```
Comment (diemtvu) [inline pilot/pkg/model/config.go]: May be you should use separate api group (e.g 'authorization'), with your actual api version + @xiaolanz

Comment (xiaolanz) [reply pilot/pkg/model/config.go]: Hey, this has to be in "rbac" and v1alpha1.

**Comment by yangminzhu** [comment]
@diemtvu Thanks for catching this! 
I just realized that the istioctl has 2 different sources of scheme definition. The validation part actually works as the readInputs() is using model.IstioConfigTypes, but other parts of istioctl is using newClient().

File: pilot/pkg/model/config.go (hunk 1 above)
Comment (yangminzhu) [reply pilot/pkg/model/config.go]: It's not a new CRD, it's currently defined [here](https://github.com/istio/istio/blob/4ac60af504902674f1452848d4110ffbaaf571e0/install/kubernetes/templates/istio-mixer.yaml.tmpl#L587) (group='config' and version='v1alpha2').

**Pull request merged**